packages/tool-cache/src/tool-cache.ts

```diff
@@ -344,21 +344,49 @@ async function extractZipWin(file: string, dest: string): Promise<void> {
   // build the powershell command
   const escapedFile = file.replace(/'/g, "''").replace(/"|\n|\r/g, '') // double-up single quotes, remove double quotes and newlines
   const escapedDest = dest.replace(/'/g, "''").replace(/"|\n|\r/g, '')
-  const command = `$ErrorActionPreference = 'Stop' ; try { Add-Type -AssemblyName System.IO.Compression.FileSystem } catch { } ; [System.IO.Compression.ZipFile]::ExtractToDirectory('${escapedFile}', '${escapedDest}')`
-
-  // run powershell
-  const powershellPath = await io.which('powershell', true)
-  const args = [
-    '-NoLogo',
-    '-Sta',
-    '-NoProfile',
-    '-NonInteractive',
-    '-ExecutionPolicy',
-    'Unrestricted',
-    '-Command',
-    command
-  ]
-  await exec(`"${powershellPath}"`, args)
+  const pwshPath = await io.which('pwsh', false)
```

Comment:
notes from discussion:
- test behavior of tar.gz (linux/macOS/win) and xar (macOS), and 7z (win) to determine whether they match. Good to know, not necessarily a show-stopper.
- i think overlay (and overwrite conflicts) is what we want. Might be good to get clarification from customer.
- pwsh on windows-latest
  - i think windows-latest has pwsh in the PATH
  - fallback case, you might need to hack the code (specify an invalid overload)
  - error case (i.e. where we put `$_` back into the pipeline), you can specify an invalid path
  - might need to add debug tracing so you know which path was taken
- powershell on windows-latest
  - rename pwsh out of the way (or delete it)